rust_src/src/buffers.rs

```diff
@@ -103,3 +103,12 @@ pub fn get_buffer(buffer_or_name: LispObject) -> LispObject {
         ))
     }
 }
+
+/// Return the current buffer as a Lisp object.
+#[lisp_fn]
+pub fn current_buffer() -> LispObject {
```

Comment:
Good question!

The C code definitely uses Fcurrent_buffer in quite a few places, but if we define a `ThreadState::current_buffer` then we can offer an interface with a better Rusty type.

There's also the question of whether or not `ThreadState::current_buffer` should be a static function, or a method on a static value `current_thread`.

I've pushed an additional commit: what do you think? I'm sure it could be prettier somehow, but I'm not sure how.